### PR TITLE
Outreachy Ceph Heatlh Monitoring Plugins Application - Email Module

### PR DESCRIPTION
**Pull request by shruthi0898**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/email/module.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    ]
+
+    def handle_command(self, inbuf, cmd):
+        self.log.info("Email_info")
+        self.log.debug("Email_debug")
+        self.log.error("Email_error")
+
+        status_code = 0
+        output_buffer = "Output buffer is for data results"
+        output_string = "Output string is for informative text"
```
Comment (liewegas) [inline src/pybind/mgr/email/module.py]: we don't really need these lines.  on success, we don't need a friendly message.. or can just put it directly in teh return statement below.

File: src/pybind/mgr/email/module.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        sender = cmd['sender_id']
+        receivers = cmd['target_id']
+
+        email_message = cmd['msg']
```
Comment (liewegas) [inline src/pybind/mgr/email/module.py]: Let's try taking the result of self.get('health') and including that content into the message.  Either dump it out (e.g., '%s' % self.get('health')), or try to iterate over it to generate a more human-readable email message that looks something like what you get from 'ceph health detail'.

**Comment by liewegas** [comment]
Hi @shruthi0898, this looks like a good start!  If you have some time to fiddle with the email content to make it include the health alerts themselves, that would be great.  We'll be looking at all the applications tomorrow to make a decision.

**Comment by shruthi0898** [comment]
Hello Sage,

I was not able to record my contribution due to network error. I just saw
that. Is there any way to rectify this?

Hoping to work with you,
Shruthi

**Comment by liewegas** [comment]
It looks like this pull request came through okay.  Whatever you have (or 
have time to work on in the next ~16 hours or so) that you can include in 
this PR will be helpful.  Or even describe what your next steps might be 
in your application.  Thanks!

**Comment by shruthi0898** [comment]
Sure, I'll make the changes here. But I was not able to complete my outreachy application and have contacted the organisers. Is there anything else I can do? 

Hoping to find a solution,
Shruthi

**Comment by shruthi0898** [comment]
Hello, 

Can you please tell a project timeline that I can give for the application?

Thanking you,
Shruthi

**Comment by liewegas** [comment]
My plan was to be flexible on the scope for the project.  There are several ways we can publish alerts: email, twitter, slack, irc, maybe even sms.  My hope is that there will be some generic infrastructure we do to track the health alerts, and then a series of 'output' modes for each way that notifications can get sent.

I would suggest a timeline that gets an initial smtp-base notification working quickly, then refine that to improve the alert structure (nicely formatted message content, notifications when alerts clear, etc.), and then time to add at least one additional output modes (irc, twitter, slack, etc).

File: src/pybind/mgr/email/module.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        if 'port' in cmd:
+            port = cmd['port']
```
Comment (sebastian-philipp) [inline src/pybind/mgr/email/module.py]:
```python
port = cmd.get('port', 465)
```
should work here, too.

**Comment by liewegas** [comment]
Thanks for your interest in Outreachy!

**Pull request closed without merging**